tests/httpwrappers/tests.py

```diff
@@ -530,6 +530,46 @@ def test_content_type_mutually_exclusive(self):
                 headers={"Content-Type": "text/csv"},
             )
 
+    def test_text_property(self):
+        test_cases = [
+            (
+                "Hello, world!",
+                "Updated content",
+                "Hello, world!",
+                "Updated content",
+                None,
+                "Initial content test failed.",
+            ),
+            (
+                b"Ol\xc3\xa1 Mundo",
+                b"Ol\xe1 Mundo",
+                "Olá Mundo",
+                "Olá Mundo",
+                "iso-8859-1",
+                "UTF-8 encoding test failed.",
+            ),
+        ]
+
+        for (
+            initial_content,
+            updated_content,
+            expected_initial,
+            expected_updated,
+            charset,
+            description,
+        ) in test_cases:
+            with self.subTest(content=initial_content):
+                response = HttpResponse(
+                    initial_content,
+                    content_type="text/plain; charset=utf-8" if charset else None,
+                )
+                self.assertEqual(response.text, expected_initial, description)
+
+                response.content = updated_content
+                if charset:
+                    response.charset = charset
+                self.assertEqual(response.text, expected_updated, description)
```

Comment:
```suggestion
    def test_text(self):
        response = HttpResponse("Hello, world!", content_type=None)
        self.assertEqual(response.text, "Hello, world!")

        response.content = "Updated content"
        self.assertEqual(response.text, "Updated content")

    def test_text_charset(self):
        for content_type, content in [
            ("text/plain; charset=utf-8", b"Ol\xc3\xa1 Mundo"),
            ("text/plain; charset=iso-8859-1", b"Ol\xe1 Mundo"),
        ]:
            with self.subTest(content_type=content_type):
                response = HttpResponse(content, content_type=content_type)
                self.assertEqual(response.text, "Olá Mundo")
```
I find this hard to read - I think we should split this into two tests

Reply:
Splitting the sections as you suggested makes it look much cleaner. Thank you for the feedback!